pkg/controller/framework/controller_test.go

```diff
@@ -371,11 +366,18 @@ func TestUpdate(t *testing.T) {
 		},
 	}
 
-	// run every test a few times, in parallel
 	const threads = 3
+	testDoneWG.Add(threads * len(tests))
```

Comment:
Nit: slight preference for moving this line up instead of moving the controller start down here.